eclipse-collections/src/main/java/org/eclipse/collections/impl/collector/SerializableDoubleSummaryStatistics.java

```diff
@@ -61,14 +63,33 @@
             max = DoubleSummaryStatistics.class.getDeclaredField("max");
             max.setAccessible(true);
         }
-        catch (SecurityException | NoSuchFieldException ignored)
+        catch (Exception e)
         {
-            count = null;
-            sum = null;
-            sumCompensation = null;
-            simpleSum = null;
-            min = null;
-            max = null;
+            if (e instanceof SecurityException || e instanceof NoSuchFieldException || e.getClass().getCanonicalName().equals("java.lang.reflect.InaccessibleObjectException"))
+            {
+                count = null;
+                sum = null;
+                sumCompensation = null;
+                simpleSum = null;
+                min = null;
+                max = null;
+            }
+            else
+            {
+                try
+                {
+                    throw e;
+                }
+                catch (NoSuchFieldException ignored)
```

Comment:
Yes. If you see previously the same exception was caught. If I remove the `throw` then it is not necessary, but it is incorrect, we should throw.
Also, it is a compiler error.